### PR TITLE
Added TF source retrieval with CMake and some formatting changes to R…

### PR DESCRIPTION
**Pull request by alxhoff**
…EADME

### CODE REVIEW

File: cmake/FindTFLite.cmake [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    ${TFL_SRCS}
+)
+
+TARGET_INCLUDE_DIRECTORIES(tflite PUBLIC
+    ${TFL_INC_DIRS}
+)
+
+TARGET_COMPILE_DEFINITIONS(tflite PUBLIC
+    TF_LITE_USE_GLOBAL_CMATH_FUNCTIONS
+    TF_LITE_STATIC_MEMORY
```
Comment (rafzi) [inline cmake/FindTFLite.cmake:156]:
```
TF_LITE_USE_GLOBAL_MAX
TF_LITE_USE_GLOBAL_MIN
```

I needed these in the latest TF when compiling for RISC-V.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,10 +1,29 @@
...
-SET(TF_DIR "../tensorflow" CACHE STRING "TensorFlow source directory")
-SET(TFL_SRC ${TF_DIR}/tensorflow/lite)
-SET(TFLM_SRC ${TFL_SRC}/micro)
-SET(TFLMD_SRC ${TFLM_SRC}/tools/make/downloads)
+IF(NOT TF_DIR)
+    SET(TF_DIR "../tensorflow" CACHE STRING "TensorFlow source directory")
+ENDIF()
+
+GET_FILENAME_COMPONENT(tf_fullpath ${TF_DIR} REALPATH)
+IF(EXISTS "${tf_fullpath}" AND NOT GET_TF_SRC)
```
Comment (rafzi) [inline CMakeLists.txt:10]: We could be more explicit and do this decision just on GET_TF_SRC. Then move the path check down and give a hard error if it doesn't exist.

File: README.md [hunk 3]
```diff
@@ -7,36 +7,87 @@ function calls assigned to the model, then dumps the tensor and node settings in
...
+## Make
 
 - check out tensorflow master next to this project (in ../tensorflow)
 - start with building the tflite micro library as described in https://www.tensorflow.org/lite/microcontrollers/library:
 
-  - cd ../tensorflow
+  - `cd ../tensorflow`
 
-  - make -f tensorflow/lite/micro/tools/make/Makefile hello_world_bin
+  - `make -f tensorflow/lite/micro/tools/make/Makefile hello_world_bin`
```
Comment (rafzi) [inline README.md:66]: This is required for the CMake non-automatic path as well.

File: cmake/FindTFLite.cmake (hunk 1 above)
Comment (cpetig) [reply cmake/FindTFLite.cmake:156]: yeah, I recall that problem

**Comment by alxhoff** [comment]
Had a day of holidays yesterday but I will finish this up now and include your changes. Cheers

**Comment by alxhoff** [comment]
Check error is because of #22

**Comment by cpetig** [comment]
@alxhoff would you mind to take a look into the failing checks? The code base should have become more stable now with the large changes and tensorflow adaptions merged.

**Comment by alxhoff** [comment]
> @alxhoff would you mind to take a look into the failing checks? The code base should have become more stable now with the large changes and tensorflow adaptions merged.

Failing checks were fixed in #22. I just packed it into a separate PR.

**Pull request merged**